asmcomp/cmm_helpers.ml

```diff
@@ -106,11 +106,17 @@ let alloc_boxedintnat_header dbg = Cconst_natint (boxedintnat_header, dbg)
 let max_repr_int = max_int asr 1
 let min_repr_int = min_int asr 1
 
+let tag_const (n : int) : nativeint =
+  Nativeint.(add (shift_left (of_int n) 1) 1n)
+
+let untag_const (n : nativeint) : int =
+  assert Nativeint.(logand n 1n = 1n);
```

Comment:
I don't like assertions like this.
I tend to prefer `Misc.fatal_error`, unless you think the assertion obviously holds given the code around it (which is not the case here, since the code that guarantees the assertion is much firther down).

Reply:
done